traits.rs

```diff
@@ -119,3 +122,16 @@ pub trait PSP22 {
         delta_value: u128,
     ) -> Result<(), PSP22Error>;
 }
+
+#[ink::trait_definition]
+pub trait PSP22Metadata {
+    /// Returns the token name.
+    #[ink(message)]
+    fn token_name(&self) -> Option<String>;
+    /// Returns the token symbol.
+    #[ink(message)]
+    fn token_symbol(&self) -> Option<String>;
+    /// Returns the token decimals.
+    #[ink(message)]
+    fn token_decimals(&self) -> u8;
+}
```

Comment:
Are you going to implement it for the `PSP22Data` struct?

Reply:
You could do it similarly to `PSP22Data` and have `struct PSP22Metadata` on which you implement the trait. It'd be consistent w/ the already-taken approach.

Reply:
That would mean building two layers of wrapper abstraction around accessing a constant ^^'

Reply:
You'll be in charge of explaining how to use this lib to the users ;)